packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -474,16 +472,41 @@ export class TabBarRenderer extends TabBar.Renderer {
         return h.div({ className: baseClassName, style }, data.title.iconLabel);
     }
 
+    protected renderExtendedTabBarPreview = (title: Title<Widget>) => {
+        const hoverBox = document.createElement('div');
+        hoverBox.classList.add('theia-horizontal-tabBar-hover-div');
+        const labelElement = document.createElement('p');
+        labelElement.classList.add('theia-horizontal-tabBar-hover-title');
+        labelElement.textContent = title.label;
+        hoverBox.append(labelElement);
+        if (title.caption) {
+            const captionElement = document.createElement('p');
+            captionElement.classList.add('theia-horizontal-tabBar-hover-caption');
+            captionElement.textContent = title.caption;
+            hoverBox.appendChild(captionElement);
+        }
+        return hoverBox;
+    };
+
     protected handleMouseEnterEvent = (event: MouseEvent) => {
         if (this.tabBar && this.hoverService && event.currentTarget instanceof HTMLElement) {
             const id = event.currentTarget.id;
             const title = this.tabBar.titles.find(t => this.createTabId(t) === id);
             if (title) {
-                this.hoverService.requestHover({
-                    content: title.caption,
-                    target: event.currentTarget,
-                    position: 'right'
-                });
+                if (this.tabBar.orientation === 'horizontal') {
+                    this.hoverService.requestHover({
+                        content: this.renderExtendedTabBarPreview(title),
+                        target: event.currentTarget,
+                        position: 'bottom',
+                        cssClasses: ['extended-tab-preview']
+                    });
```

Comment:
This might be a bit untransparent, but this is catched by https://github.com/eclipsesource/theia/blob/69bb4dfa86d5dc39bc39b35e95fb6bca4fc99e43/packages/core/src/browser/shell/tab-bars.ts#L160-L162.

We do not call this the `handleMouseEnterEvent` for horizontal tab bars if the setting is not set to true.

Reply:
I've noticed some issues where we display the past hover on preference update that might be interesting to investigate:


https://user-images.githubusercontent.com/40359487/228247056-7bbb9158-3b5d-4e8d-bef3-cc1605333295.mp4

Reply:
Good catch! I will investigate it.

Reply:
I added a listener to the preference change that triggers an update on all tab bars. Now the correct hover should be displayed directly after changing the preference.
Additionally i made it so that the css classes are removed after a hover was disposed. This way the sidebar hovers will not apply the same css classes.

(Also added the localization)

Could you take another look?